client/src/www/app/outline_server_repository/access_key.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {SHADOWSOCKS_URI} from 'ShadowsocksConfig';
+
+import {TunnelConfig, TransportConfig} from './vpn';
+import * as errors from '../../model/errors';
+
+// Parses an access key string into a TunnelConfig object.
+export function staticKeyToTunnelConfig(staticKey: string): TunnelConfig {
+  try {
+    const config = SHADOWSOCKS_URI.parse(staticKey);
+    return {
+      transport: new TransportConfig({
+        host: config.host.data,
+        port: config.port.data,
+        method: config.method.data,
+        password: config.password.data,
+        prefix: config.extra?.['prefix'],
+      }),
+    };
+  } catch (cause) {
+    throw new errors.ServerAccessKeyInvalid('Invalid static access key.', {
+      cause,
+    });
+  }
+}
+
+export function validateStaticKey(staticKey: string) {
+  let config = null;
+  try {
+    config = SHADOWSOCKS_URI.parse(staticKey);
```

Comment:
Currently `staticKeyToTunnelConfig()` doesn't do much validation.
This is something that still needs to be cleaned up, but I'd rather leave it to another PR, since this one is already large.